src/csharp/Grpc.Core/Internal/GrpcThreadPool.cs

```diff
@@ -165,11 +170,19 @@ private void RunHandlerLoop(CompletionQueueSafeHandle cq, IProfiler optionalProf
                     try
                     {
                         var callback = cq.CompletionRegistry.Extract(tag);
-                        callback(success);
+                        // Use cached delegates to avoid unnecessary allocations
+                        if (!inlineHandlers)
+                        {
+                            ThreadPool.QueueUserWorkItem(success ? RunCompletionQueueEventCallbackSuccess : RunCompletionQueueEventCallbackFailure, callback);
```

Comment:
I just tried with Task.Run() and the performance seems about the same, but the advantage of ThreadPool.QueueUserWorkItem is that it saves allocations, which might become interesting in the future when we look into optimizing the performance a bit more (I already know about some other places where we allocate unnecessarily and I've tested that removing those allocations will improve the performance).